api/model/src/testUriCompliance/java/org/projectnessie/model/TestContentKeyEncoding.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.model;
+
+import static java.lang.String.format;
+import static org.eclipse.jetty.http.UriCompliance.Violation.ILLEGAL_PATH_CHARACTERS;
+import static org.eclipse.jetty.http.UriCompliance.Violation.SUSPICIOUS_PATH_CHARACTERS;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.eclipse.jetty.http.HttpURI;
+import org.eclipse.jetty.http.UriCompliance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@ExtendWith(SoftAssertionsExtension.class)
+public class TestContentKeyEncoding {
+  @InjectSoftAssertions protected SoftAssertions soft;
+
+  @ParameterizedTest
+  @MethodSource
+  public void fromPathString(
+      String encoded,
+      List<String> elements,
+      Collection<UriCompliance.Violation> expectedViolations) {
+    List<String> actualElements = Util.fromPathString(encoded);
+    soft.assertThat(actualElements).containsExactlyElementsOf(elements);
+    soft.assertThat(actualElements).containsExactlyElementsOf(Util.fromPathString(encoded));
+
+    HttpURI uri = HttpURI.from(format("http://hostname/%s", encoded));
+    soft.assertThat(uri.getViolations())
+        .describedAs("path-beginning: %s", encoded)
+        .containsExactlyInAnyOrderElementsOf(expectedViolations);
```

Comment:
I'm not sure it's a sufficient test.... e.g. when `#` is part of `encoded` it would be treated as a fragment by the URI, thus corrupting key data. I think we have to URI-encode `#` then run it through `HttpURI`, check violations, then get the path and parse the path from `HttpURI`. WDYT?

Reply:
I think you're overthinking this. `#` for example is escaped using `UrlEncoder`.

Reply:
maybe... too much context switching :sweat_smile: I'll try it locally :hourglass:

Reply:
reviewing today (hopefully)

Reply:
added a separate comment, resolving this thread.